android/app/src/main/java/labs/pooh/eaterslab/repository/CafeteriasRepository.kt

```diff
@@ -0,0 +1,82 @@
+package labs.pooh.eaterslab.repository
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import labs.pooh.client.apis.CafeteriasApi
+import labs.pooh.client.apis.FixedMenuReviewsApi
+import labs.pooh.client.models.FixedMenuOptionReview
+import labs.pooh.eaterslab.repository.dao.*
+import labs.pooh.eaterslab.ui.activity.abstracts.ConnectionStatusNotifier
+import org.threeten.bp.OffsetDateTime
+import java.lang.Exception
+
+class CafeteriasRepository(private  val connectionStatusNotifier: ConnectionStatusNotifier) {
+
+    private val cafeteriasApi by lazy { CafeteriasApi() }
+    private val reviewsApi by lazy { FixedMenuReviewsApi() }
+
+    private val cachedCafeterias = mutableMapOf<Int, CafeteriaDao>()
+    private val cachedMenuOptions = mutableMapOf<Int, Set<FixedMenuOptionDao>>()
```

Comment:
A List instead of set? I think options should be preserved in order returned from the server. Now it seems that hashing implementation decides ordering.

Reply:
will fix this, good point of view